src/gdk/gdk_bat.c

```diff
@@ -374,6 +383,43 @@ BATattach(int tt, const char *heapfile, int role)
 		p = Tloc(bn, 0);
 		n = (lng) st.st_size;
 		while (n > 0 && (m = fread(p, 1, (size_t) MIN(1024*1024, n), f)) > 0) {
+#if defined(HAVE_EMBEDDED) && defined(HAVE_EMBEDDED_JAVA)
```

Comment:
Why swap? Is the endianness in the JVM different?

Reply:
The JVM is always big endian. My machine is little endian and I was getting problems with the binary import.